spec/integration/api/likes_controller_spec.rb

```diff
@@ -0,0 +1,177 @@
+        like_service(auth.user).create(@status.guid)
+        like_service(alice).create(@status.guid)
+        get(
+          api_v1_post_likes_path(post_id: @status.guid),
+          params: {access_token: access_token}
+        )
+        expect(response.status).to eq(200)
+        likes = response_body(response)
+        expect(likes.length).to eq(3)
+        confirm_like_format(likes[0], alice)
+        confirm_like_format(likes[1], bob)
+        confirm_like_format(likes[2], auth.user)
+      end
+    end
+
+    context "with wrong post id" do
+      it "fails at getting likes" do
+        get(
+          api_v1_post_likes_path(post_id: "badguid"),
+          params: {access_token: access_token}
+        )
+        expect(response.status).to eq(404)
+        expect(response.body).to eq(I18n.t("api.endpoint_errors.posts.post_not_found"))
+      end
+    end
+  end
+
+  describe "#create" do
+    context "with right post id" do
+      it "succeeeds in liking post" do
+        post(
+          api_v1_post_likes_path(post_id: @status.guid),
+          params: {access_token: access_token}
+        )
+        expect(response.status).to eq(204)
+        likes = like_service.find_for_post(@status.guid)
+        expect(likes.length).to eq(1)
+        expect(likes[0].author.id).to eq(auth.user.person.id)
+      end
+
+      it "fails in liking already liked post" do
+        post(
+          api_v1_post_likes_path(post_id: @status.guid),
+          params: {access_token: access_token}
+        )
+        expect(response.status).to eq(204)
+
+        post(
+          api_v1_post_likes_path(post_id: @status.guid),
+          params: {access_token: access_token}
+        )
+        expect(response.status).to eq(422)
+        expect(response.body).to eq(I18n.t("api.endpoint_errors.likes.like_exists"))
+
+        likes = like_service.find_for_post(@status.guid)
+        expect(likes.length).to eq(1)
+        expect(likes[0].author.id).to eq(auth.user.person.id)
+      end
+    end
+
+
+    context "with wrong post id" do
+      it "fails at liking post" do
+        post(
+          api_v1_post_likes_path(post_id: 99_999_999),
+          params: {access_token: access_token}
+        )
+        expect(response.status).to eq(404)
+        expect(response.body).to eq(I18n.t("api.endpoint_errors.posts.post_not_found"))
+      end
+    end
+  end
+
+  describe "Unlike a post" do
+    before do
+      post(
+        api_v1_post_likes_path(post_id: @status.guid),
+        params: {access_token: access_token}
+      )
+    end
+
+    context "with right post id" do
+      it "succeeds at unliking post" do
+        delete(
+          api_v1_post_likes_path(post_id: @status.guid),
+          params: {access_token: access_token}
+        )
+        expect(response.status).to eq(204)
+        likes = like_service.find_for_post(@status.guid)
+        expect(likes.length).to eq(0)
+      end
+
+      #TODO Re-enable test if decided by team this is correct behavior
```

Comment:
- Layout/LeadingCommentSpace: Missing space after `#`.
- Style/CommentAnnotation: Annotation keywords like `TODO` should be all upper case, followed by a colon, and a space, then a note describing the problem.